src/openms/source/ANALYSIS/MAPMATCHING/QTClusterFinder.cpp

```diff
@@ -99,10 +111,166 @@ namespace OpenMS
     // update parameters (dummy)
     setParameters_(1, 1);
 
+    if(use_IDs_)
+    {
+      // map string "modified sequence/charge" to all RTs the feature has been observed in the different maps
+      std::unordered_map<String, std::vector<double>> ided_feat_rts;
+      //std::unordered_map<String, std::vector<const typename MapType::FeatureType*>> ided_feats;
+      double minRT = std::numeric_limits<double>::max();
+      for (const auto& map : input_maps)
+      {
+        for (const auto& feat : map)
+        {
+          if (feat.getRT() < minRT) minRT = feat.getRT();
+          const auto& pepIDs = feat.getPeptideIdentifications();
+          if (!pepIDs.empty())
+          {
+            const auto& hits = pepIDs[0].getHits();
+            //TODO sort (lose const) or assume first is best? Especially since we sort later in run_internal_ anyway
+            if (!hits.empty())
+            {
+              if ((hits[0].getScore() > min_score_ && pepIDs[0].isHigherScoreBetter()) ||
+                  (hits[0].getScore() < min_score_ && !pepIDs[0].isHigherScoreBetter()))
+              {
+                //TODO we could loosen the score filtering by requiring only ONE ID of a peptide to pass the threshold.
```

Comment:
what does this mean in the case that e.g., 20 PSMs with same peptide are mapped to the feature?

Reply:
I intend to only use the best per feature. But if there are multiple features that have a best ID that maps to this peptide it could be enough if one exceeds the score to include all their differences. Just an idea.